src/main/scala/uk/gov/nationalarchives/aggregate/processing/AggregateProcessingLambda.scala

```diff
@@ -104,6 +116,76 @@ class AggregateProcessingLambda extends RequestHandler[SQSEvent, Unit] {
     } yield AssetProcessingResult(assetProcessingErrors, suppliedMetadata)
   }
 
+  private def createMetadataCSV(assetProcessingResults: List[AssetProcessing.AssetProcessingResult]): File = {
```

Comment:
I would abstract this out into a separate scala object to the "utilies" package: `CSVUtils`